tests/utils/test_modeling_utils.py

```diff
@@ -894,32 +895,42 @@ def test_from_pretrained_low_cpu_mem_usage_functional(self):
     @require_usr_bin_time
     @require_accelerate
     @mark.accelerate_tests
-    def test_from_pretrained_low_cpu_mem_usage_measured(self):
-        # test that `from_pretrained(..., low_cpu_mem_usage=True)` uses less cpu memory than default
+    def test_from_pretrained_low_cpu_mem_usage_slower(self):
```

Comment:
Shouldn't it be faster? 

```suggestion
    def test_from_pretrained_low_cpu_mem_usage_faster(self):
```

Reply:
Was running the tests, it's a bit slower due to the added hooks IMO (which is fine, as `low_cpu_mem_usage=True` is still needed 99% of the time, aka when weights != precision)

Reply:
Hi @muellerzr,

Thanks for your work! I'm really interested in understanding the changes made on the code level. Can you tell me what is "model precision" and "weights precision". I don't get the difference. I am aware of weights being FP32, BF16 etc.

Reply:
Model precision relates to the precision the model is initialized in. When using `AutoModel` that is the `torch_dtype` param.

Weight precision is what the actual *weights* of the model are stored in.

For instance, with `transformers` when we do fine-tuning one should always train with `float32` architecture ideally, then use autocast to train in `bfloat16` so that users can make the best use of your weights and have stable fine-tuning.

*However*, models like `llama-3-8B` only release `bfloat16` weights. 

So given this, if we choose to create the architecture in `bfloat16` as well (again, in this case ideally just for inference), then we can see this fast speedup as the architecture weights are the same as the pretrained weights.

Does this make sense?